hera_pspec/uvpspec_utils.py

```diff
@@ -100,12 +101,92 @@ def subtract_uvp(uvp1, uvp2, run_check=True, verbose=False):
     if run_check: uvp1.check()
     return uvp1
 
+def compress_r_params(r_params_dict):
+    """
+    Convert a dictionary of r_paramsters to a compressed string format
+
+    Parameters
+    ----------
+    r_params_dict: Dictionary
+              dictionary with parameters for weighting matrix. Proper fields
+              and formats depend on the mode of data_weighting.
+              data_weighting == 'sinc_downweight':
+                            dictionary with fields
+                            'filter_centers', list of floats (or float) specifying the (delay) channel numbers
+                                              at which to center filtering windows. Can specify fractional channel number.
+                            'filter_widths', list of floats (or float) specifying the width of each
+                                             filter window in (delay) channel numbers. Can specify fractional channel number.
+                            'filter_factors', list of floats (or float) specifying how much power within each filter window
+                                              is to be suppressed.
+    Returns
+    -------
+    string containing r_params dictionary in json format and only containing one
+    copy of each unique dictionary with a list of associated baselines.
+    """
+    if r_params_dict == {} or r_params_dict is None:
+        return ''
+    else:
+        r_params_unique = {}
+        r_params_unique_bls = {}
+        r_params_index = -1
+        for rp in r_params_dict:
+            #do not include data set in tuple key
+            already_in = False
+            for rpu in r_params_unique:
+                if r_params_unique[rpu] == r_params_dict[rp]:
+                    r_params_unique_bls[rpu] += [rp,]
+                    already_in = True
+            if not already_in:
+                r_params_index += 1
+                r_params_unique[r_params_index] = copy.copy(r_params_dict[rp])
+                r_params_unique_bls[r_params_index] = [rp,]
+        for rpi in r_params_unique:
+            r_params_unique[rpi]['baselines'] = r_params_unique_bls[rpi]
+        r_params_str = json.dumps(r_params_unique)
+        return r_params_str
+
+def decompress_r_params(r_params_str):
+    """
+    Decompress json format r_params string into an r_params dictionary.
+
+    Parameters
+    ----------
+    r_params_str: String
+        string with compressed r_params in json format
+
+    Returns
+    -------
+    dictionary with parameters for weighting matrix. Proper fields
```

Comment:
Should follow proper format for parameter docstrings. Otherwise it breaks readthedocs!
e.g.
```
Returns
-----------
r_params : dict
    Dictionary with...
```

Reply:
This should be fixed now :).